### ISSUE TITLE
table-layout vs. update-array

### ISSUE THREAD

**Issue by bloukrans** [avenirer/CodeIgniter-MY_Model#243]
previous version would allow arrays "bigger" than the table, e.g. ignore extra fields of submit-buttons.
after upgrade i have to manually unset($updateArray['submitBtn']) otherwise it will throw

MY_Model: Unknown column (submitBtn) in table: (test).

config error on my side?

thanks!

**Comment by salain** [comment]
No it is not a config error on your part. 
if you look a PR #210 , _prep_before_write now throw an error if a the update/insert array contains an unknown column as before it simply ignored it. 
So you now have to clean up your post array before or you could use from_form as it will clean up the array for you. However if you use from_form you MUST have a validation rule for every fields otherwise any missing field will be remove. 
TIP: for any field that do not require specific validation just use trim as a validation rule.

**Comment by bloukrans** [comment]
Thanks for the fast revert and detailed explanation!

In my case this is really a bummer - i'm providing RESTful services on top of MY_MODEL - so from_form is quite a detour. How about enabling / disabling this feature by config switch?

**Comment by salain** [comment]
Based on the fact you have a submit button in your data array, I will assume you are making a POST, so from_form would work as this method uses/check the input->post array and remove protected and unknown fields. 
If this does not suite your needs you can always override "_prep_before_write" method in the specific model, to remove the error code for unknown fields. 

I think a config variables for errors would be a nightmare.

**Comment by bloukrans** [comment]
yes ajax POST with formdata but i don't create the frontend with CI and validation happens outside as well  so in this case would be double-define validation...

not adding a config variables for errors - 
just thinking of a switch e.g. line 131: protected $allow_validation_ignore = true;
and in line 257, 272 just adding "if (!$allow_validation_ignore)" before  show_error

did it locally now, would be great if this could merge

**Comment by salain** [comment]
Even if the frontend is not CI, the post is made to the CI environment, so the input post would contain your formdata. As for validation to work with from_form you only need to have trim as a rule for every field you want added/updated.
For your suggestion,I would say, what you re suggesting is specific to your need as well as incomplete. 
I think if you want to suppress errors it should be globally not only for what you need/want. 

I think if an error suppression flag is added it should apply to all errors as well as to be set using the CI_ENV constant, meaning if CI_ENV is "production" suppress all errors. 

@avenirer , what do you think?

But, for your case the override would be the best approach for now. As it is simple and quick.

**Comment by bloukrans** [comment]
Hi,

CI_env production sounds good although I already have a workaround for it. 
can you explain what is the downside of the previous behaviour of the model?
security issues?

**Comment by salain** [comment]
Obviously, previously the model would simply dismiss any field in the update/insert array that were not in the table, without any errors. This was undesirable when developing/debugging, as misspelled field name were removed without warning.
